src/core/observables/BondAngles.hpp

```diff
@@ -36,13 +36,15 @@ class BondAngles : public PidObservable {
   using PidObservable::PidObservable;
 
   std::vector<double>
-  evaluate(Utils::Span<const Particle *const> particles) const override {
+  evaluate(Utils::Span<std::reference_wrapper<const Particle>> particles,
+           const GenObs::traits<Particle> &traits) const override {
```

Comment:
Maybe declare a ParticleRefSpan for Utils::Span<std::reference_wrapper<const Particle>> ?